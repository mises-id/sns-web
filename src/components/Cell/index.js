/*
 * @Author: lmk
 * @Date: 2021-07-15 12:58:47
 * @LastEditTime: 2021-08-29 12:33:10
 * @LastEditors: lmk
 * @Description: cell Component
 */
import React from 'react';
import Image from '../Image';
import arrow from '@/images/arrow.png'
import './index.scss'
/**
 * @description: cell params introduce
 * @param {*} icon show iconImage
 * @param {*} label show cell txt
 * @param {*} iconSize iconSize
 * @param {*} rightChild right element
 * @param {*} className custom className
 * @param {*} showIcon isshow icon
 * @param {*} showLine isshow bottomLine
 * @return {*} cellelement
 */
const Cell = ({icon,label,onPress,iconSize,showArrow=true,rightChild,className="m-padding-tb20",showIcon=true,showLine=true,labelStyle={},shape="circle"})=>{
  return <div className={`m-flex m-row-between  ${className} ${showLine ? 'm-line-bottom' : ''}`} onClick={onPress}>
  <div className="m-flex">
    {showIcon&&<Image size={iconSize} shape={shape} source={icon}></Image>}
    <span className={`m-colors-333 m-font16 ${showIcon ? 'm-margin-left13' : ''}`} style={labelStyle}>{label}</span>
  </div>
  <div className="m-flex">
    {rightChild&&<div className={`${showArrow ? 'm-margin-right15' : ''}`}>{rightChild}</div>}
    {showArrow&&<img src={arrow} alt="" className="arrow"/>}
  </div>
</div>
}
export default Cell