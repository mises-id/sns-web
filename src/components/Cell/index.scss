.arrow{
  width: 5px;
}