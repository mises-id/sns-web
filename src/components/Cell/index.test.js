/*
 * @Author: lmk
 * @Date: 2021-07-15 10:12:21
 * @LastEditTime: 2021-07-15 13:46:58
 * @LastEditors: lmk
 * @Description: cell test
 */
import React from 'react';
import {shallow} from 'enzyme';
import Cell from './';
describe('Cell', () => {
  it('should render correctly', () => {
    const props = {
      showIcon:false,
      label:'test',
      iconSize:20
    }
    const component = shallow(<Cell {...props}/>)
    expect(component).toMatchSnapshot();
  });
});