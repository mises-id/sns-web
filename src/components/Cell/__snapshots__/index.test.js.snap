// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell should render correctly 1`] = `
<div
  className="m-flex m-row-between  m-padding-tb20 m-line-bottom"
>
  <div
    className="m-flex"
  >
    <div
      className=""
    >
      <span
        className="m-colors-333 m-font16"
        style={Object {}}
      >
        test
      </span>
      <div
        className="m-margin-top5"
      />
    </div>
  </div>
  <div
    className="m-flex right-content"
  >
    <img
      alt=""
      className="arrow"
      src="arrow.png"
    />
  </div>
</div>
`;
